lib/web/apiserver.go

```diff
@@ -282,6 +283,30 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 			return
 		}
 
+		// check proxy and auth node versions (breaking change from v5).
+		// If there is a mismatch with the current proxy client, redirect users to an error page to update proxy.
+		//
+		// DELETE IN 5.0: this block only serves to notify users of v4.3+ to upgrade to v5.0.0
+		if strings.HasPrefix(r.URL.Path, "/web/newuser") {
+			res, err := h.auth.Ping(context.TODO())
```

Comment:
i would refactor it to a separate method  `shouldNotifyAboutV5xChanges`

Reply:
```go
if strings.HasPrefix(r.URL.Path, "/web/newuser" &&  shouldNotifyAboutV5xChanges() ) {
   ...
   http.Redirect(w, r, pathToError.String(), http.StatusFound)
   return;
}
```

Reply:
I originally put it in a separate method, but I thought it would be easier to remove if the code was in one block. What do you think?

Reply:
sounds good to me